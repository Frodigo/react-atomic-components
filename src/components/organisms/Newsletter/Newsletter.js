import React from 'react';
import PropTypes from 'prop-types';
import { useForm } from 'react-hook-form';

import Input from '../../atoms/Input/Input';
import Button from '../../atoms/Button/Button';
import Heading from '../../atoms/Text/Heading';
import Paragraph from '../../atoms/Text/Paragraph';

import classes from './Newsletter.module.css';

const Newsletter = (props) => {
    const {
        title,
        subtitle,
        submitUrl,
        footer,
        onSubmit,
        token,
        thankYouPageUrl
    } = props;
    const { register, errors, handleSubmit } = useForm();

    return (
        <form
            className={classes.newsletter}
            onSubmit={handleSubmit(onSubmit)}
            method="post"
            action={submitUrl}
            id="newsletter"
        >
            <div className={classes.newsletterContent}>
                <Heading type="h3" classes={classes.title}>
                    {title}
                </Heading>
                <Paragraph>{subtitle}</Paragraph>
                <div className={classes.fieldset}>
                    <Input
                        placeholder="Your name"
                        name="first_name"
                        errorMessage={errors ? errors.name : {}}
                        register={register('first_name', {
                            required:true
                        })}
                    />
                    <Input
                        placeholder="Your email"
                        name="email"
                        errorMessage={errors ? errors.email : {}}
                        type="email"
                        register={register({
                            required: 'Please enter your email.'
                        })}
                    />
                </div>

                <input type="hidden" name="campaign_token" value={token} />
                <input
                    type="hidden"
                    name="thankyou_url"
                    value={thankYouPageUrl}
                />
                <input type="hidden" name="start_day" value="0" />

                <footer className={classes.footer}>
                    {/* eslint-disable-next-line react/no-danger */}
                    <p dangerouslySetInnerHTML={{ __html: footer }} />
                    <Button type="submit">Subscribe</Button>
                </footer>
            </div>
        </form>
    );
};

Newsletter.propTypes = {
    footer: PropTypes.string,
    onSubmit: PropTypes.func,
    submitUrl: PropTypes.string.isRequired,
    subtitle: PropTypes.string,
    thankYouPageUrl: PropTypes.string,
    title: PropTypes.string.isRequired,
    token: PropTypes.string.isRequired
};

export default Newsletter;
