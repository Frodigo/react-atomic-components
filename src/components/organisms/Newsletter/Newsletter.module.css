.newsletter {
    background-color: #d8d8d8;
    padding: var(--spacing-xxl) var(--spacing-md);
    text-align: center;
}

.fieldset {
    margin-top: var(--spacing-md);
}

.title {
    margin-bottom: var(--spacing-xs);
}

.footer {
    margin-bottom: var(--spacing-md);
}

@media (min-width: 768px) {
    .newsletter {
        padding: var(--spacing-xxl) var(--spacing-xl);
    }

    .newsletterContent {
        max-width: var(--tablet-container-width);
        margin: 0 auto;
    }

    .fieldset {
        display: flex;
    }

    .fieldset > div {
        flex: 1 1 50%;
        padding: 0 var(--spacing-md);
    }

    .footer {
        max-width: 400px;
        margin: var(--spacing-md) auto;
    }

   .footer button {
       margin-left: auto;
       margin-right: auto;
   }
}