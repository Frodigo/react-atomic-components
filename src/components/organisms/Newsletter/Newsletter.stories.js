import React from 'react';

import { default as NewsletterComponent } from './Newsletter';

export default {
    title: 'Organisms',
    component: NewsletterComponent
};

const Template = (args) => <NewsletterComponent {...args} />;

export const Newsletter = Template.bind({});

Newsletter.args = {
    title: 'Subscribe to the newsletter',
    subtitle: 'I will keep you informed about each new blog entry',
    footer:
        'Also i will send you send you PDF with 112 pages about PWA Studio Rich Content renderers',
    // eslint-disable-next-line no-console
    onSubmit: (data) => console.log(data)
};
