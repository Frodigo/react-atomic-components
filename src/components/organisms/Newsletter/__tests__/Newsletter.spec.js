import React from 'react';
import { render } from '@testing-library/react';

import Newsletter from '../Newsletter';

describe('Newsletter organism', () => {
    const args = {
        title: 'Subscribe to the newsletter',
        subtitle: 'I will keep you informed about each new blog entry',
        submitUrl: '#',
        token: 'abc',
        footer:
            'Also i will send you send you PDF with 112 pages about PWA Studio Rich Content renderers'
    };

    test('Newsletter renders correctly', () => {
        const { container } = render(<Newsletter {...args} />);

        expect(container).toMatchSnapshot();
    });
});
