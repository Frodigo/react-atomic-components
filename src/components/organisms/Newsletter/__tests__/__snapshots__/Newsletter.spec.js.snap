// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Newsletter organism Newsletter renders correctly 1`] = `
<div>
  <form
    action="#"
    class="newsletter"
    id="newsletter"
    method="post"
  >
    <div
      class="newsletterContent"
    >
      <h3
        class="title"
      >
        Subscribe to the newsletter
      </h3>
      <p
        class="paraghaph"
      >
        I will keep you informed about each new blog entry
      </p>
      <div
        class="fieldset"
      >
        <div
          class="field"
        >
          <input
            class="input null"
            data-testid="input"
            name="first_name"
            placeholder="Your name"
            type="text"
            value=""
          />
        </div>
        <div
          class="field"
        >
          <input
            class="input null"
            data-testid="input"
            name="email"
            placeholder="Your email"
            type="email"
            value=""
          />
        </div>
      </div>
      <input
        name="campaign_token"
        type="hidden"
        value="abc"
      />
      <input
        name="thankyou_url"
        type="hidden"
        value=""
      />
      <input
        name="start_day"
        type="hidden"
        value="0"
      />
      <footer
        class="footer"
      >
        <p>
          Also i will send you send you PDF with 112 pages about PWA Studio Rich Content renderers
        </p>
        <button
          class="primary"
          type="submit"
        >
          Subscribe
        </button>
      </footer>
    </div>
  </form>
</div>
`;
